Kuadrant/kuadrant-controller
# fetch operations from virtualservice

This PR:
- Makes `name` field in the `rule` object a fetching trigger for operation from VirtualSerivce.
- Makes `rule` object take array of `Operation`s.
- Implements detach part of signaling.

## Verification Steps
1. `make local-setup`
2. `k apply -f examples/toystore/toystore.yaml`
3. `k apply -f examples/toystore/virtualService.yaml`
4. `k apply -f config/samples/apim_v1alpha1_ratelimitpolicy.yaml`
5. `k annotate virtualservice/toystore kuadrant.io/ratelimitpolicy=toystore`
6. `istioctl dashboard envoy <kuadrant-gateway-pod>.kuadrant-system`

You should be able to find the following config in the `postauth.wasm` filter which is fetched from `VirtualService` and not present in `RateLimitPolicy`.

```json
{
          "operation": [
            {
              "paths": [
                "/admin/toy"
              ],
              "methods": [
                "POST"
              ],
              "hosts": [
                "*.toystore.com"
              ]
            },
            {
              "paths": [
                "/admin/toy"
              ],
              "methods": [
                "DELETE"
              ],
              "hosts": [
                "*.toystore.com"
              ]
            }
          ],
          "actions": [
            {
              "generic_key": {
                "descriptor_value": "yes",
                "descriptor_key": "admin"
              }
            }
          ]
        }
```
